packages/kms-local/src/key-management-system.ts

```diff
@@ -68,21 +78,69 @@ export class KeyManagementSystem extends AbstractKeyManagementSystem {
-  async signEthTX({ key, transaction }: { key: IKey; transaction: object }): Promise<string> {
-    return sign(transaction, '0x' + key.privateKeyHex)
-  }
-
-  async signJWT({ key, data }: { key: IKey; data: string | Uint8Array }): Promise<string> {
+  async sign({
+    key,
+    alg,
+    data,
+    extras,
+  }: {
+    key: IKey
+    alg?: string
+    data: Uint8Array
+    extras?: KMSSignerExtras
+  }): Promise<string> {
+    //FIXME: KMS implementation should not rely on private keys being provided, but rather manage their own keys
     if (!key.privateKeyHex) throw Error('No private key for kid: ' + key.kid)
 
-    if (key.type === 'Ed25519') {
+    if (key.type === 'Ed25519' && (typeof alg === 'undefined' || ['Ed25519', 'EdDSA'].includes(alg))) {
       const signer = EdDSASigner(key.privateKeyHex)
-      return (signer(data) as any) as string
+      const signature = await signer(data)
+      //base64url encoded string
+      return signature as string
     } else if (key.type === 'Secp256k1') {
-      const signer = ES256KSigner(key.privateKeyHex)
-      return (signer(data) as any) as string
-    } else {
-      throw Error('Cannot sign JWT with key of type ' + key.type)
+      if (typeof alg === 'undefined' || ['ES256K', 'ES256K-R'].includes(alg)) {
+        const signer = ES256KSigner(key.privateKeyHex, alg === 'ES256K-R')
+        const signature = await signer(data)
+        //base64url encoded string
+        return signature as string
+      } else if (['eth_signTransaction', 'signTransaction', 'signTx'].includes(alg)) {
+        const {v, r, s, type, ...tx} = parse(data)
+        const wallet = new Wallet(key.privateKeyHex)
+        const signedRawTransaction = await wallet.signTransaction(<TransactionRequest>tx)
+        //HEX encoded string, 0x prefixed
+        return signedRawTransaction
+      } else if (alg === 'eth_signMessage') {
+        const wallet = new Wallet(key.privateKeyHex)
+        const signature = await wallet.signMessage(data)
+        //HEX encoded string, 0x prefixed
+        return signature
+      } else if (['eth_signTypedData', 'EthereumEip712Signature2021'].includes(alg)) {
+        let msg, msgDomain, msgTypes
+        const serializedData = toUtf8String(data)
+        let jsonData = JSON.parse(serializedData)
+        if (typeof jsonData.domain === 'object' && typeof jsonData.types === 'object') {
+          const { domain = undefined, types = undefined, message = undefined } = { ...extras, ...jsonData }
+          msg = message || jsonData
+          msgDomain = domain
+          msgTypes = types
+        }
+        if (typeof msgDomain !== 'object' || typeof msgTypes !== 'object') {
+          throw Error(`invalid_arguments: Cannot sign typed data. 'domain' and 'types' must be provided`)
+        }
+        const wallet = new Wallet(key.privateKeyHex)
+
+        const signature = await wallet._signTypedData(msgDomain, msgTypes, msg)
+        //HEX encoded string
+        return signature
+      }
```

Comment:
Can we split this into separate methods for better readability and testability?

Reply:
done